ls1intum/Artemis
# `Plagiarism checks`: Include only student submissions

### Checklist
#### General
#### Server

#### Changes affecting Programming Exercises

### Motivation and Context
Fixes #6023.

From the issue:

Tutors/Instructors sometimes test the code of students to replicate possible issues with the test cases. Their submissions shouldn't be considered in the plagiarism checks. Otherwise, we will have cases with high similarity, which would confuse when instructors deny them. 

### Description
This PR introduces a filter which excludes submissions from tutors, editors and instructors from plagiarism checks. The change applies to manual and continuous checks.  

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Tutor
- 2 Students (⚠️ Those students cannot be instructors not tutors in the course!)
- 1 Exercise (text, programming or modeling)

1. Create two identical submissions from two student accounts.
2. Create an identical submission from a tutor account. 
3. Create an identical submission from an instructor account.
4. As instructor run plagiarism checks for the exercise.
5. Verify that only student submissions are visible in checks results.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress
#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage

Note: _Newly added logic is tested._

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| ProgrammingPlagiarismDetectionService.java | 65% | ✅                           |
| ModelingPlagiarismDetectionService.java | 92% | ✅                           |
